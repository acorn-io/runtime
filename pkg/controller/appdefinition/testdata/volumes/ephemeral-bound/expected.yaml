kind: Deployment
apiVersion: apps/v1
metadata:
  name: container-name
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "container-name"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "container-name"
      "acorn.io/managed": "true"
  template:
    metadata:
      labels:
        "acorn.io/app-namespace": "app-namespace"
        "acorn.io/app-name": "app-name"
        "acorn.io/app-public-name": "app-name"
        "acorn.io/container-name": "container-name"
        "acorn.io/managed": "true"
      annotations:
        acorn.io/container-spec: '{"dirs":{"/var/tmp":{"secret":{},"volume":"foo"}},"image":"image-name","probes":null}'
    spec:
      hostname: container-name
      imagePullSecrets:
        - name: container-name-pull-1234567890ab
      terminationGracePeriodSeconds: 5
      enableServiceLinks: false
      serviceAccountName: container-name
      volumes:
        - name: foo
          persistentVolumeClaim:
            claimName: foo-bind
      containers:
        - name: container-name
          image: "image-name"
          volumeMounts:
            - mountPath: "/var/tmp"
              name: foo
---
kind: PodDisruptionBudget
apiVersion: policy/v1
metadata:
  name: container-name
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "container-name"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "container-name"
      "acorn.io/managed": "true"
  maxUnavailable: 25%
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: "foo-bind"
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    "acorn.io/volume-name": "foo"
    "acorn.io/volume-class": "custom-class"
    acorn.io/public-name: app-name.foo
spec:
  accessModes:
    - ReadWriteOnce
  volumeName: "existing-foo"
  resources:
    requests:
      storage: 5M
---
kind: Secret
apiVersion: v1
metadata:
  name: container-name-pull-1234567890ab
  namespace: app-created-namespace
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
type: "kubernetes.io/dockerconfigjson"
data:
  ".dockerconfigjson": eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
---
kind: AppInstance
apiVersion: internal.acorn.io/v1
metadata:
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
  volumes:
    - target: foo
      volume: existing-foo

status:
  namespace: app-created-namespace
  appImage:
    id: test
  appSpec:
    containers:
      container-name:
        image: "image-name"
        dirs:
          "/var/tmp":
            volume: foo
    volumes:
      foo:
        class: ephemeral
        size: 10
  conditions:
    - type: defined
      reason: Success
      status: "True"
      success: true
