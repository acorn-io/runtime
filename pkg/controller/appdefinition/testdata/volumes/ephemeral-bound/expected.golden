`apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
  name: container-name-pull-1234567890ab
  namespace: app-created-namespace
type: kubernetes.io/dockerconfigjson

---
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: container-name
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: container-name
  namespace: app-created-namespace

---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: container-name
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: container-name
  namespace: app-created-namespace
spec:
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: container-name
      acorn.io/managed: "true"
  strategy: {}
  template:
    metadata:
      annotations:
        acorn.io/container-spec: '{"dirs":{"/var/tmp":{"secret":{},"volume":"foo"}},"image":"image-name","metrics":{},"probes":null}'
        karpenter.sh/do-not-evict: "true"
      creationTimestamp: null
      labels:
        acorn.io/app-name: app-name
        acorn.io/app-namespace: app-namespace
        acorn.io/app-public-name: app-name
        acorn.io/container-name: container-name
        acorn.io/managed: "true"
        acorn.io/project-name: app-namespace
    spec:
      containers:
      - image: image-name
        name: container-name
        resources: {}
        volumeMounts:
        - mountPath: /var/tmp
          name: foo
      enableServiceLinks: false
      hostname: container-name
      imagePullSecrets:
      - name: container-name-pull-1234567890ab
      serviceAccountName: container-name
      terminationGracePeriodSeconds: 10
      volumes:
      - name: foo
        persistentVolumeClaim:
          claimName: foo-bind
status: {}

---
apiVersion: policy/v1
kind: PodDisruptionBudget
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: container-name
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: container-name
  namespace: app-created-namespace
spec:
  maxUnavailable: 25%
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: container-name
      acorn.io/managed: "true"
status:
  currentHealthy: 0
  desiredHealthy: 0
  disruptionsAllowed: 0
  expectedPods: 0

---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/managed: "true"
    acorn.io/public-name: other-app.foo
    acorn.io/volume-class: custom-class
    acorn.io/volume-name: foo
  name: foo-bind
  namespace: app-created-namespace
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5M
  storageClassName: foo
  volumeName: existing-foo
status: {}

---
apiVersion: internal.acorn.io/v1
kind: AppInstance
metadata:
  creationTimestamp: null
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
  volumes:
  - target: foo
    volume: other-app.foo
status:
  appImage:
    buildContext: {}
    id: test
    imageData: {}
    vcs: {}
  appSpec:
    containers:
      container-name:
        dirs:
          /var/tmp:
            secret: {}
            volume: foo
        image: image-name
        metrics: {}
        probes: null
    volumes:
      foo:
        class: ephemeral
        size: 10G
  appStatus: {}
  columns: {}
  conditions:
    reason: Success
    status: "True"
    success: true
    type: defined
  defaults: {}
  namespace: app-created-namespace
  staged:
    appImage:
      buildContext: {}
      imageData: {}
      vcs: {}
  summary: {}
`
